src/utils/token/tokenFetch.tsx

```diff
@@ -35,31 +34,40 @@ export const fetchAllTokenAssets = (): Token[] => {
   return tokens;
 };
 
+export function useTokenAssets() {
+  const envs = useEnvContext();
+  return fetchAllTokenAssets(envs['PENUMBRA_CHAIN_ID'] ?? '');
+}
```

Comment:
As this is, I'm pretty sure that fetchAllTokenAssets() does not cache. If env context was in react-query, we could compose within react-query to handle caching globally for this work. Otherwise, every component's use of useTokenAssets() will be a new instance and re-running on every render.